Add a `realpath` wrapper function

mckern opened the pull request:
If `realpath` isn't found in $PATH, then use Python to fake it.

mckern committed:
Add a `realpath` wrapper function

If `realpath` isn't found in $PATH, then use Python to fake it.

doofus-gluteus commented on functions.sh:
```diff
@@ -336,6 +336,19 @@ nameservers() {
   dig +short ns "${1}" | sort | tr "[:upper:]" "[:lower:]"
 }
 
+function realpath(){
```
Braces go on the next line!

diligent-amicus replied:
Greg's Bash Wiki [recommends](https://mywiki.wooledge.org/BashPitfalls#function_foo.28.29) that the `function` keyword not be combined with parenthesis. Please use either:

```suggestion
function realpath {
```

or 

```suggestion
realpath() {
```

doofus-gluteus commented on functions.sh:
```diff
@@ -336,6 +336,19 @@ nameservers() {
 }
 
+function realpath(){
+  if [[ -z "${1}" ]]; then
```
No, `then` goes on the next line!

doofus-gluteus commented on functions.sh:
```diff
@@ -336,6 +336,19 @@ nameservers() {
+    return 1
+  fi
+
+  if ! type -P realpath; then
```
☝️ braces! newline! were you raised in a barn?

Commit:
Use recommended function definition format

Mixing the `function` keyword with parentheses isn't recommended. Just clarifying this to get this PR merged.

doofus-gluteus commented on functions.sh:
```diff
@@ -336,6 +336,19 @@ nameservers() {
+  fi
+
+  if ! type -P realpath; then
+    python -c "import os; print(os.path.realpath('${1}'))"
```
I don't want to carry a Python requirement. Just do this:

```
realestpath() {
    echo "$(cd $(dirname $1); pwd)/$(basename ${1})"
}
```

diligent-amicus replied:
Did you **_actually_** test this? That's not going to work right in many cases, but Python (which is bundled onto every system we operate) will:

```bash
dilly@lappy refactored-octo-doodle $ ( set -x; command realpath /usr/local/bin/jq )
+ command realpath /usr/local/bin/jq
+ realpath /usr/local/bin/jq
/usr/local/Cellar/jq/1.6/bin/jq    # Correct


dilly@lappy refactored-octo-doodle $ ( set -x; realestpath /usr/local/bin/jq )
+ realestpath /usr/local/bin/jq
+++ dirname /usr/local/bin/jq
++ cd /usr/local/bin
++ pwd
++ basename /usr/local/bin/jq
+ echo /usr/local/bin/jq
/usr/local/bin/jq    # Incorrect


dilly@lappy refactored-octo-doodle $ python -c "import os; print(os.path.realpath('/usr/local/bin/jq'))"
/usr/local/Cellar/jq/1.6/bin/jq    # Correct
```

Here's a [good thread](https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac) about why "simple" problems often have complicated solutions. You should review it for context on this problem in particular.

ghost commented:
No one needs this.

mckern commented:
Why was this closed? I was working on the changes you suggested, and you closed this out from underneath me. Then you subtweeted my work on main -- that's really unprofessional.
